release/models/sampling/openconfig-sampling-sflow.yang

```diff
@@ -222,25 +274,36 @@ module openconfig-sampling-sflow {
         "Enables or disables sFlow sampling for the device.";
     }
 
-    leaf source-address {
-      type oc-inet:ip-address;
+    leaf agent-id-ipv4 {
+      type oc-inet:ipv4-address;
       description
-        "Sets the source IP address for sFlow datagrams sent to
-        sFlow collectors.";
+        "Sets the agent identifier for IPv4 PDUs.";
+      reference
+        "RFC 3176 - InMon Corporation's sFlow: A Method for
+        Monitoring Traffic in Switched and Routed Networks";
     }
 
-    leaf sampling-rate {
-      type uint32;
+    leaf agent-id-ipv6 {
+      type oc-inet:ipv6-address;
       description
-        "Sets the global packet sampling rate.  The rate is
-        is expressed as an integer N, where the intended sampling
-        rate is 1/N packets.  An implementation may implement the
-        sampling rate as a statistical average, rather than a strict
-        periodic sampling.
-
-        The allowable sampling rate range is generally a
-        property of the system, e.g., determined by the
-        capability of the hardware.";
+        "Sets the agent identifier for IPv6 PDUs.";
+      reference
+        "RFC 3176 - InMon Corporation's sFlow: A Method for
+        Monitoring Traffic in Switched and Routed Networks";
+    }
+
+    leaf source-address-ipv4 {
```

Comment:
That is correct - defined globally for all collectors

Reply:
SGTM.

Reply:
@robshakir - Rethinking this one a bit - I've relocated the source addresses out of global configuration to each collector instance and aggregated under a single leaf node of type `oc-yang:ip-address`.  Since each collector in the model is network-instance aware, a single source address at the global level does not make much sense.

Commit: https://github.com/openconfig/public/pull/522/commits/ad4199f774e9037100f3921e1e1304932ec161bf

Tree excerpt:
```
module: openconfig-sampling
  +--rw sampling
     +--rw oc-sflow:sflow
        +--rw oc-sflow:collectors
           +--rw oc-sflow:collector* [address port]
              +--rw oc-sflow:config
                 +--rw oc-sflow:address?            oc-inet:ip-address
                 +--rw oc-sflow:port?               oc-inet:port-number
                 +--rw oc-sflow:source-address?     oc-inet:ip-address
                 +--rw oc-sflow:network-instance?   oc-netinst:network-instance-ref
```

Reply:
Nice catch, probably the most common scenario is an sflow collector would use a source ip in a management vrf.